src/components/datepicker/datepicker-component.jsx

```diff
@@ -0,0 +1,469 @@
+import React, { useState } from 'react';
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { DayPicker, useDayPicker } from 'react-day-picker';
+import { format } from 'date-fns';
+import { cn } from '@/utilities/functions';
+import 'react-day-picker/style.css';
+import Button from '../button';
+import { currentTimeDot, formatWeekdayName, generateYearRange } from './utils';
+
+const DatePickerComponent = ( {
+	width,
+	className: outerClassName, // Renamed to avoid shadowing
+	classNames,
+	selectedDates,
+	setSelectedDates,
+	showOutsideDays = false,
+	mode = 'single',
+	variant = 'normal',
+	alignment = 'horizontal',
+	onBlur,
+	onChange,
+	...props
+} ) => {
+	//set state for month and year selection
+	const numberOfMonths = props.numberOfMonths;
```

Comment:
We can get (destructure) it from the props above instead of taking a new variable.

Reply:
done